tests/shell/tests/01-run.py

```diff
@@ -54,6 +54,11 @@
     ('\b\b\b\becho', '\"echo\"'),
     ('help', EXPECTED_HELP),
     ('echo a string', '\"echo\"\"a\"\"string\"'),
+    ("""echo "t\e st" "\\"" '\\'' a\ b""", '"echo""te st"""""\'""a b"'),
```

Comment:
I think there missing test cases that are not clear to me how they should be handled:

- `echo asdfas"asdfasdf`
- `echo asdfas'asdfasdf`
- `echo asd"fas'asdfa"sdf`

Reply:
Note that these where not handled before either.

Reply:
As a note in my terminal it results in:
- `echo asdfas"asdfasdf`: handled as unterminated message
- `echo asdfas'asdfasdf`: handled as unterminated message
- `echo asd"fas'asdfa"sdf`: echoes `asdfas'asdfasdf`
- `echo asd"fasasdfa"sdf`: echoes `asdfasasdfasdf`

Reply:
I think everything works as expected. My outputs are:
(1) echo abcdef"ghijklmn => "echo""abcdef"ghijklmn"
(2) echo abcdef'ghijklmn => "echo""abcdef'ghijklmn"
(3) echo abc"def'ghijk"lmn => "echo""abc"def'ghijk"lmn"
(4) echo abc"defghijk"lmn => "echo""abc"defghijk"lmn"

The problem for (1), (3) and (4) is that the implemented `echo` command does not care about escaping `"`. A more clear output should probably look like this:
(1) echo abcdef"ghijklmn => "echo""abcdef\"ghijklmn"
(3) echo abc"def'ghijk"lmn => "echo""abc\"def'ghijk\"lmn"
(4) echo abc"defghijk"lmn => "echo""abc\"defghijk\"lmn"
Therefore, the `echo` command needs to escape `"`. I believe I should add this.

Reply:
> The problem for (1), (3) and (4) is that the implemented echo command does not care about escaping ". A more clear output should probably look like this:

But don't you think that (1) and (2) should be considered invalid messages? its not different from:

`echo "asdfasdf` or `"echo asdfasdf'"`

Reply:
The difference is that the quote is in the beginning vs the quote is within the current token. In former case we are in state `PARSE_SPACE`, in the latter in state `PARSE_UNQUOTED`. Currently we can reach the quoting state only from `PARSE_SPACE`. I'm not sure what is common here, but I know that at least bash would allow us to start quoting within a token.

Reply:
I will change the code such that there is a transition from `PARSE_UNQUOTED` to `PARSE_SINGLEQUOTE` and `PARSE_DOUBLEQUOTE` and back.

Reply:
This way your observations will be the same for the terminal and RIOT's shell.